clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1505,9 +1506,27 @@ public void resume(Collection<TopicPartition> partitions) {
      */
     @Override
     public void close() {
+        close(DEFAULT_CLOSE_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Tries to close the consumer cleanly within the specified timeout. This method waits up to
+     * <code>timeout</code> for the consumer to complete pending commits and leave the group.
+     * If the consumer is unable to complete the requests before the timeout expires, the consumer
+     * is force closed.
```

Comment:
@apurvam Thank you for the review. Yes, that is a good point.

I was trying to make the behaviour no worse than it was before. For auto-commit, the code was waiting indefinitely to find the coordinator. I changed that to timeout. For `commitAsync`, the code was not waiting for coordinator. I didn't want that to block for 10 minutes in the default case. But now that the default timeout is a much more reasonable 30 seconds, I think it makes sense for the code to be consistent in both cases. So I have modified `close` to wait up to timeout to find coordinator when commits are pending. The alternative would be not to wait in either case, but that would be worse behaviour for auto-commit than it is now.

@hachikuji The changes are in the latest commit. Hope they make sense. Thank you.